Fix CSVLoader to use specified parameter when estimating number of columns

kmaehashi opened the pull request:
`CSVLoader` has a feature to automatically assign field names for each column ([c1, c2, c3, ...]) based on the number of columns in the first row of the CSV file.  However, specified constructor options (`*args` and `**kwargs`, in which users may specify delimiter, dialect, etc.) are not used when counting number of columns.  This PR fixes the problem.

rimms commented:
In Python 2.7,  ``fieldnames=False`` does not work when using non-ascii file.
My test is the followings:
```
  def test_noheader_non_ascii(self):
    with TempFile() as f:
      f.write("いち|に|さん\n1|2|3\n".encode('utf-8'))
      f.flush()
      loader = CSVLoader(f.name, fieldnames=False, encoding='utf-8', delimiter='|')
```

Error Messages:
```
Traceback (most recent call last):
  File "/home/rimms/jubakit/jubakit/test/loader/test_csv.py", line 62, in test_noheader_non_ascii
    loader = CSVLoader(f.name, fieldnames=False, encoding='utf-8', delimiter='|')
  File "/home/rimms/jubakit/jubakit/loader/csv.py", line 55, in __init__
    fieldnames = ['c{0}'.format(i) for i in range(len(r.fieldnames))]
  File "/home/rimms/.pyenv/versions/2.7.9/lib/python2.7/csv.py", line 90, in fieldnames
    self._fieldnames = self.reader.next()
UnicodeEncodeError: 'ascii' codec can't encode characters in position 0-1: ordinal not in range(128)
```

In Python 3.5, this patch works fine.

kmaehashi commented:
@rimms Thank you for review.
The root cause of the original issue was that we used different mechanism to (1) detect number of columns and (2) actually load the data.
I fixed this issue by extracting csv-related things into ``_UnicodeDictReader`` class, so now  (1) and (2) share the same code.

I also added several tests against CP932 files.

rimms commented:
👍 LGTM
